lib/utils/storage_directory_utils.dart

```diff
@@ -34,4 +40,38 @@ class StorageDirectoryUtils {
         await StorageDirectoryUtils.instance.getDownloadFolderInApp();
     return '$downloadInAppFolder/$eventId/$fileName';
   }
+
+  Future<String?> getTwakeDownloadsFolderInDevice() async {
+    try {
+      final downloadPath = await ExternalPath.getExternalStoragePublicDirectory(
+        ExternalPath.DIRECTORY_DOWNLOADS,
+      );
+      if (downloadPath.isNotEmpty == true) {
+        return '$downloadPath/${AppConfig.applicationName}';
+      }
+      throw SaveToDownloadsException(error: 'Download path is empty');
+    } catch (e) {
+      Logs().e('StorageDirectoryUtils:: getDownloadFolderDevice: $e');
+    }
+    return null;
+  }
+
+  Future<String> getAvailableFilePath(String filePath) async {
+    String availableFilePath = filePath;
+    final positionOfDot = filePath.lastIndexOf('.');
+    String extension = '';
+    String fileName = '';
+    int counter = 1;
+    if (positionOfDot != -1) {
+      extension = filePath.substring(positionOfDot);
+      fileName = filePath.substring(0, positionOfDot);
+    } else {
+      fileName = filePath;
+    }
+    while (await File(availableFilePath).exists()) {
+      availableFilePath = '$fileName ($counter)$extension';
+      counter++;
+    }
+    return availableFilePath;
+  }
```

Comment:
what is it? can we write test for it?

Reply:
its not easy to write test for this file, because it depends on the files of users have